frontend/web/components/pages/GitHubSetupPage.tsx

```diff
@@ -0,0 +1,245 @@
+import React, { FC, useState, useEffect } from 'react'
+import OrganisationSelect from 'components/OrganisationSelect'
+import Input from 'components/base/forms/Input'
+import Icon from 'components/Icon'
+import InputGroup from 'components/base/forms/InputGroup'
+import ProjectFilter from 'components/ProjectFilter'
+import Utils from 'common/utils/utils'
+import Button from 'components/base/forms/Button'
+import { useCreateGithubIntegrationMutation } from 'common/services/useGithubIntegration'
+import { useCreateGithubRepositoryMutation } from 'common/services/useGithubRepository'
+import { useGetGithubReposQuery } from 'common/services/useGithub'
+import PanelSearch from 'components/PanelSearch'
+
+type Location = {
+  search: string
+}
+
+type GitHubSetupPageType = {
+  location: Location
+}
+
+type projectType = {
+  value: string
+  label: string
+}
+
+type repoType = {
+  value: string
+  label: string
+  name?: string
+}
+
+const GitHubSetupPage: FC<GitHubSetupPageType> = (props) => {
```

Comment:
I don't think this should be a separate page, it should either be part of the integrations page or project settings.

Reply:
I think even if we did have a separate page, this looks a bit unfinished? 

<img width="1511" alt="image" src="https://github.com/Flagsmith/flagsmith/assets/8608314/81d3a2c4-016a-490d-94cf-e5038f950c96">

Reply:
Managed to get here with an installation id but see a 500 pointing to the api branch url, I think that should work unless environment vars need adding ? 
<img width="1168" alt="image" src="https://github.com/Flagsmith/flagsmith/assets/8608314/a116c10d-3095-4cd8-a4b9-6c7a82b459b1">

Reply:
>I don't think this should be a separate page, it should either be part of the integrations page or project settings. If you check out Vercel for example as a good flow for connecting GH.
Testing this out it has to work more like that anyway, since if you've already connected to GitHub there's no callback url, you just see the ability to configure.

The ability to install the app without leaving the app has been added. However, the old flow will take you to the setup page if you want to install it from GitHub.

> Managed to get here with an installation id but see a 500 pointing to the api branch url, I think that should work unless environment vars need adding ?

Yes, a var env named GITHUB_PEM is necessary, which is described in the API PR.